app/src/main/java/com/infomaniak/mail/data/models/message/Message.kt

```diff
@@ -228,20 +229,18 @@ class Message : RealmObject, Snoozable {
                     scope.setExtra("email", AccountUtils.currentMailboxEmail.toString())
                 }
 
-                correctFolder = (threads.firstOrNull { it.folderId == folderId } ?: threads.first()).folder
+                correctFolder = (threads.firstOrNull { it.folder.id == folderId } ?: threads.first()).folder
```

Comment:
Also, between `Thread.folder.id` and `Thread.folderId`, which one is the correct one that should be used and why?

Reply:
Both are supposed to be ok, but things don't seem to really work here, so I'm trying stuff. 😭

Reply:
Ok but then you won't understand the reason why `it.folderId == folderId` wasn't working then, because the code will have changed and might not capture a message anymore.

I don't know how much better you expect this new solution to work but if you really want this version instead, maybe try to compute the old solution in parallel and send sentry about the old version even if the code fallsback to the new solution. Or/and, maybe send to sentry differences when you detect that `thread.folderId != thread.folder.id` for any thread

Reply:
Actually I reproduce the issue on master very easily when navigating the search. You should probably start looking though this first instead to understand the issue. No need to add these sentries, for now, to understand your issue.